frontend/pages/queries/QueryPage/QueryPage.jsx

```diff
@@ -104,6 +106,18 @@ export class QueryPage extends Component {
     return false;
   }
 
+  componentDidMount() {
+    const { dispatch, relatedHost } = this.props;
+
+    // A fetch call is required for the host data if we are coming in new to the app
+    // and are interested in the query in the context of a single host
+    // e.g. coming from link /queries/new?host_ids=4
+    if (relatedHost !== undefined) {
+      const { fetchHost } = helpers;
+      fetchHost(dispatch, relatedHost);
+    }
+  }
```

Comment:
Is there a reasonable way to avoid the extra request this generates if the host is already loaded? If it's going to add complexity I'm fine with leaving as-is.

Reply:
yep. reworked the code to optimise the network requests. It'll only request the host if the data is not already on the client now.